HPCC-29584 Add github action to test helm/k8s

jpmcmu opened the pull request:
- Added a new action to install an HPCC cluster on K8s via Helm
- Added a new base regression suite on top of K8s to build-vcpkg

Signed-off-by: James McMullan James.McMullan@lexisnexis.com

<!-- Thank you for submitting a pull request to the HPCC project

 PLEASE READ the following before proceeding.

 This project only accepts pull requests related to open JIRA issues.
 If suggesting a new feature or change, please discuss it in a JIRA issue first.
 If fixing a bug, there should be an issue describing it with steps to reproduce.
 The title line of the pull request (and of each commit within it) should refer to the
 associated issue using the format:

 HPCC-nnnnn Short description of issue

 This will allow the Jira ticket to be automatically updated to refer to this pull request,
 and will ensure that the automatically-generated changelog is properly formatted.
 Where a pull request contains a single commit the pull request title will be set automatically,
 assuming that the commit has followed the proper guidelines.

 Please go over all the following points, and put an `x` in all the boxes that apply. You may find
 it easier to press the 'Create' button first then click on the checkboxes to edit the comment.
-->

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [X] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [X] My code follows the code style of this project.
  - [X] My code does not create any new warnings from compiler, build system, or lint.
- [X] The commit message is properly formatted and free of typos.
  - [X] The commit message title makes sense in a changelog, by itself.
  - [X] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [X] I have read the CONTRIBUTORS document.
- [X] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [X] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [X] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

## Testing:
<!-- Please describe how this change has been tested.-->

<!-- Thank you for taking the time to submit this pull request and to answer all of the above-->

jakesmith reviewed (commented):
@jpmcmu - looks great, a few minor questions, and I'd like @GordonSmith to look it over too.

jpmcmu commented on .github/actions/deploy-hpcc-k8s/action.yaml:
```diff
@@ -0,0 +1,178 @@
+      run: |
+        k8s_pkg_path=$(ls -t ./${{ inputs.asset-name }}/*.deb 2>/dev/null | head -1)
+        k8s_pkg_file=$(basename "$k8s_pkg_path")
+        mv ${k8s_pkg_path} ${{ inputs.platform-folder }}/${k8s_pkg_file}
```
Good question, I believe I was doing that to debug an issue and can probably just reference the original path directly.

jpmcmu replied:
@jakesmith Turns out this was required. The Dockerfile looks for the pkg file in the platform directory. I added a comment noting why it was required.

jpmcmu commented on .github/actions/deploy-hpcc-k8s/action.yaml:
```diff
@@ -0,0 +1,178 @@
+        sleep 10
+        echo "Waiting for ECLWatch startup" && kubectl wait --for=condition=ready pod --timeout=180s -l app=eclwatch
+        echo "Waiting for Rowservice startup" && kubectl wait --for=condition=ready pod --timeout=180s -l server=rowservice
+        echo "Waiting for SQL2ECL startup" && kubectl wait --for=condition=ready pod --timeout=180s -l app=sql2ecl
```
I have some tests that I will be adding in a subsequent PR that utilize those services, but it would be better to wait on all deployments to finish. I will make that change.

jakesmith commented on .github/workflows/test-regression-suite-k8s.yml:
```diff
@@ -0,0 +1,53 @@
+    steps:
+      - shell: "bash"
+        run: |
+          echo "...all tests passed..."
```
Right, it will be good to get this foundation merged, which will test k8s startup etc, then we can start to expand to add regression tests.

jpmcmu commented:
@GordonSmith implemented code review changes, please take a look when you get a chance.

jpmcmu commented:
@jakesmith Implemented code review changes, please review

jakesmith reviewed (approved):
@jpmcmu - looks good.
The "Check that eclhelper interface has not changed / Check eclhelper interface (pull_request)" step is failing, but no idea why, that file hasn't changed, despite what the step details suggest. Could it have accidentally in a push version of this PR?

I'll force a rerun to see if it clears.

jpmcmu committed:
HPCC-29584 Add github action to test helm/k8s

- Added a new action to install an HPCC cluster on K8s via Helm
- Added a new base regression suite on top of K8s to build-vcpkg

Signed-off-by: James McMullan James.McMullan@lexisnexis.com
